### PR TITLE
Add fields to `getgovproposal`

### PR DESCRIPTION
**Pull request by hidiego**
#### What kind of PR is this?:

/kind fix

#### What this PR does / why we need it:

Adds missing fields to `getgovproposal` and `listgovproposals`

### CODE REVIEW

**Review by Jouzo — DISMISSED** [review]

File: src/masternodes/rpc_proposals.cpp [hunk 1]
```diff
@@ -3,8 +3,27 @@
...
+    uint32_t approvalThreshold{0};
+    switch(prop.type) {
+        case CPropType::CommunityFundProposal:
+            approvalThreshold = attributes->GetValue(cfpMajority, Params().GetConsensus().props.cfp.approvalThreshold) / 10000;
+            break;
+        case CPropType::VoteOfConfidence:
+            approvalThreshold = attributes->GetValue(vocMajority, Params().GetConsensus().props.voc.approvalThreshold) / 10000;
+            break;
+    }
+
```
Comment (Jouzo) [inline src/masternodes/rpc_proposals.cpp:26]: Can use `GetMajorityFromAttributes` and `GetQuorumFromAttributes` (which handles `vocEmergencyQuorum` as well)

Comment (Jouzo) [reply src/masternodes/rpc_proposals.cpp:26]: Should rename `GetMajorityFromAttributes` to `GetApprovalThresholdFromAttributes` at the same time

File: src/validation.cpp [hunk 2]
```diff
@@ -4268,10 +4268,10 @@ void CChainState::ProcessProposalEvents(const CBlockIndex* pindex, CCustomCSView
 
         switch(prop.type) {
             case CPropType::CommunityFundProposal:
-                majorityThreshold = attributes->GetValue(cfpMajority, chainparams.GetConsensus().props.cfp.majorityThreshold) / 10000;
+                majorityThreshold = attributes->GetValue(cfpMajority, chainparams.GetConsensus().props.cfp.approvalThreshold) / 10000;
```
Comment (Jouzo) [inline src/validation.cpp:4271]: Same as above, we can use helpers `GetQuorumFromAttributes` and `GetMajorityFromAttributes` in validation

Comment (hidiego) [reply src/validation.cpp:4271]: fixing in next commit with further improvements

File: src/masternodes/masternodes.cpp [hunk 3]
```diff
@@ -1340,22 +1340,27 @@ CAmount CCustomCSView::GetMajorityFromAttributes(const CPropType& type) const
...
-    CDataStructureV0 QuorumKey{AttributeTypes::Governance, GovernanceIDs::Proposals, GovernanceKeys::Quorum};
+    CDataStructureV0 quorumKey{AttributeTypes::Governance, GovernanceIDs::Proposals, GovernanceKeys::Quorum};
+    CDataStructureV0 vocEmergencyQuorumKey{AttributeTypes::Governance, GovernanceIDs::Proposals, GovernanceKeys::VOCEmergencyQuorum};
 
-    return attributes->GetValue(QuorumKey, Params().GetConsensus().props.quorum);
+    if (type == CPropType::VoteOfConfidence && emergency) {
+        return attributes->GetValue(vocEmergencyQuorumKey, COIN / 10) / 10000;
+    } else {
+        return attributes->GetValue(quorumKey, Params().GetConsensus().props.quorum) / 10000;
+    }
```
Comment (prasannavl) [inline src/masternodes/masternodes.cpp:1363]:
```suggestion
    if (type == CPropType::VoteOfConfidence && emergency) {
        return attributes->GetValue(vocEmergencyQuorumKey, COIN / 10) / 10000;
    }
    return attributes->GetValue(quorumKey, Params().GetConsensus().props.quorum) / 10000;
```

File: src/masternodes/proposals.h [hunk 4]
```diff
@@ -114,8 +111,6 @@ struct CPropObject : public CCreatePropMessage {
         READWRITE(creationHeight);
         READWRITE(proposalEndHeight);
         READWRITE(votingPeriod);
-        READWRITE(majority);
-        READWRITE(quorum);
```
Comment (prasannavl) [inline src/masternodes/proposals.h:118]: Just a note - requires reindex from start of v3 testnet fork.

File: src/masternodes/rpc_proposals.cpp [hunk 5]
```diff
@@ -648,7 +689,10 @@ UniValue listgovproposals(const JSONRPCRequest& request)
         if (type && type != uint8_t(prop.type)) {
             return true;
         }
-        ret.push_back(propToJSON(propId, prop));
+        if(cycle != 0 && prop.cycle != cycle){
```
[the remark this answers is not in the document]
Comment (prasannavl) [reply src/masternodes/rpc_proposals.cpp:692]: Let's just do cycle = 0 /1 for now and skip the rest for v3.0

**Review by prasannavl — CHANGES_REQUESTED** [review]

File: src/masternodes/rpc_proposals.cpp (hunk 5 above)
[the remark this answers is not in the document]
Comment (hidiego) [reply src/masternodes/rpc_proposals.cpp]: We already have status for getting only "Voting" proposals so there is no need for cycle as we were planning.

**Pull request merged**